### PR TITLE
FIX Connector Implementation

### PR DESCRIPTION
**Pull request by colinpattison**
Initial commit of the FIX connector implementation including BME example script.

- all submodules updated to latest
- added loggedOffEvent handling and waitForLogoff
- added sendCancel, sendModify with mapping of fields to all connectors

C

### CODE REVIEW

File: examples/bme-cdr-example.cpp [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+        std::string e("failed to configure logger: " + errorMessage);
+        errx (1, "%s", e.c_str ());
+    }
+
+    properties props (p, "gwc", "millennium", "sim");
+    props.setProperty ("host", "127.0.0.1:9899");
+    props.setProperty ("begin_string", "FIXT.1.1");
+    props.setProperty ("sender_comp_id", "DBL");
+    props.setProperty ("target_comp_id", "BME");
+    props.setProperty ("data_dictionary", "/home/colinp/dev/blu-corner/fosdk/examples/FIXT11.xml");
```
Comment (richardfaloon) [inline examples/bme-cdr-example.cpp]: Could this be a relative path?

File: examples/bme-cdr-example.cpp [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+    order.setInteger (AccountType, 1);
+    order.setString (Symbol, "CSCO");
+    order.setSide (GWC_SIDE_BUY);
+    order.setQty (100);
+    order.setOrderType (GWC_ORDER_TYPE_LIMIT);
+    order.setString (ReceivedDeptID, "T");
+
+    cdrDateTime tt;
+    getTime (tt);
+    order.setDateTime (TransactTime, tt);
```
Comment (richardfaloon) [inline examples/bme-cdr-example.cpp]: I think we should make the sendOrder etc fill in the TransactTime?

File: src/fix/gwcFix.h [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+#include "sbfMw.h"
+
+#include "fixCodec.h"
+
+#include <map>
+
+using namespace std;
+using namespace neueda;
+
+extern const string FixHeartbeat;
```
Comment (richardfaloon) [inline src/fix/gwcFix.h:19]: Why are you defining extern references here?

File: src/fix/gwcFix.h [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+                       gwcMessageCallbacks* messageCbs, 
+                       const neueda::properties& props);
+
+    virtual bool start (bool reset);
+
+    virtual bool stop ();
+
+    virtual bool traderLogon (string& traderId, const cdr* msg = NULL)
+    {
+        /* no trader logon */
```
Comment (richardfaloon) [inline src/fix/gwcFix.h]: Exchanges like ICE are FIX and need a traderLogon

File: src/fix/gwcFix.h [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+    virtual bool sendCancel (cdr& cancel);
+    virtual bool sendCancel (gwcOrder& cancel);
+    virtual bool sendModify (cdr& modify);
+    virtual bool sendModify (gwcOrder& modify);
+    virtual bool sendMsg (cdr& msg);
+    virtual bool sendRaw (void* data, size_t len);
+
+protected:
+    SbfTcpConnection*             mTcpConnection;
+    gwcFixTcpConnectionDelegate mTcpConnectionDelegate;
```
Comment (richardfaloon) [inline src/fix/gwcFix.h:101]: Style nit, spacing doesn't line up

File: src/fix/gwcFix.cpp [hunk 6]
```diff
@@ -0,0 +1,938 @@
+#include "gwcFix.h"
+#include "sbfInterface.h"
+#include "utils.h"
+#include "fields.h"
+
+#include <sstream>
+
+extern const string FixHeartbeat = "0";
```
Comment (richardfaloon) [inline src/fix/gwcFix.cpp]: These don't need to be extern they are not used outside this compilation uint are they?

File: examples/bme-cdr-example.cpp (hunk 1 above)
Comment (colinpattison) [reply examples/bme-cdr-example.cpp]: Aye - I'll switch to using the util function for getting it from an environment variable path

**Comment by colinpattison** [comment]
I've addressed all of the above except traderLogon. I've raised #17 to track that. As discussed I don't think that matches with the rest of the interface so probably needs tweaked

**Pull request merged**